src/main/java/com/internet/shop/security/AuthenticationServiceServiceImpl.java

```diff
@@ -16,10 +17,11 @@
     public User login(String login, String password)
             throws AuthenticationException {
         Optional<User> userFromDB = userService.findByLogin(login);
-        if (userFromDB.isEmpty()
-                || !userFromDB.get().getPassword().equals(password)) {
-            throw new AuthenticationException("Incorrect login or password!");
+        if (userFromDB.isPresent()
+                && (userFromDB.get().getPassword()
+                .equals(HashUtil.hashPassword(password, userFromDB.get().getSalt())))) {
```

Comment:
```suggestion
                && passwordIsValid(userFromDB.get(), password)) {
```
let's create a private method here